### DISCUSSION TITLE
Add / view sqlite generated columns (computed columns)

### DISCUSSION THREAD

**Discussion by topvis** [dbeaver/dbeaver#17529] [Q&A]
sqlite generated columns ([ref](https://www.sqlite.org/gencol.html#:~:text=1.-,Introduction,can%20not%20be%20directly%20written.)) don't show under "Tables->Columns". Is generated column supported by DBeaver?
If the generated column is created in other software and the data will show in DBeaver correctly. It's just that DBeaver won't show the generated column info or allow adding generated columns.

**Comment by mkgrgis** [comment]
It seems there is support of generated columns for PostgreSQL. I think Your question will transform to feature request for adaptation a part of pg functional to SQLite.

**Comment by LonwoLonwo** [comment]
Hello @topvis 

Can you privide me table DDL for this case:

> If the generated column is created in other software and the data will show in DBeaver correctly. 

For now I can't see generated columns in the table with the sample DDL

```sql
CREATE TABLE t1(
   a INTEGER PRIMARY KEY,
   b INT,
   c TEXT,
   d INT GENERATED ALWAYS AS (a*abs(b)) VIRTUAL,
   e TEXT GENERATED ALWAYS AS (substr(c,b,b+1)) STORED
);
```

So it is interesting to find cases, then generated columns is shown.

Also, that driver version do you use?

**Comment by topvis** [comment]
@LonwoLonwo 
My test table DDL is
```SQL
CREATE TABLE TestTabel (
	k TEXT PRIMARY KEY,
	firstName TEXT,
	lastName TEXT,
	fullName TEXT AS (firstName || ' ' || lastName)
);

```
And the generated columns won't show up in the navigation panel but the data of the generated column does show.
![image](https://user-images.githubusercontent.com/5996531/188151509-d7ebd05b-af86-451f-b379-9a48a8a52513.png)



The driver version I'm using is sqlite-jdbc 3.39.2.1

**Reply by topvis** [reply]
Looks it's fixed with the upgraded SQLite driver (version 3.39.2.1) https://github.com/xerial/sqlite-jdbc/releases

**Reply by LonwoLonwo** [reply]
Great news!
